deploy/contracts/AddressProxy.sol

```diff
@@ -90,13 +90,13 @@ contract AddressProxy {
     }
 
     function isNewAddress(bytes32 id) internal view returns (bool) {
-      bool isNew = true;
-      for (uint i = 0; i < availableIds.length; i++) {
-        if(id == availableIds[i]) {
-          isNew = false;
+        bool isNew = true;
+        for (uint i = 0; i < availableIds.length; i++) {
```

Comment:
Wouldn't it be more efficient to check if the id exists in the mapping? Given the default value for a non existing key in the mapping is a value with byte-representation as zeros. [Solidity docs on mappings](http://solidity.readthedocs.io/en/v0.4.21/types.html#mappings)